modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/MeResourceManagerTest.java

```diff
@@ -0,0 +1,1547 @@
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(ENDPOINT_ME);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(AbstractResourceManager.encodeSCIMException(any(ConflictException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new CharonException()));
+        when(userManager.createMe(anyObject(), anyObject())).thenThrow(CharonException.class);
+
+        SCIMResponse outputScimResponse = meResourceManager.create(scimObjectString, userManager,
+                attributes, excludeAttributes);
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+    }
+
+    @DataProvider(name = "dataForTestDeleteSuccess")
+    public Object[][] dataToTestDeleteSuccess()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        User user = getNewUser();
+        String username = user.getUsername();
+        return new Object[][]{
+                {username, ResponseCodeConstants.CODE_NO_CONTENT}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestDeleteSuccess")
+    public void testDeleteSuccess(String userName, int expectedScimResponseStatus)
+            throws NotFoundException {
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(ENDPOINT_ME);
+
+        SCIMResponse outputScimResponse = meResourceManager.delete(userName, userManager);
+
+        //Assertions
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+    }
+
+    @DataProvider(name = "dataForTestDeleteFails")
+    public Object[][] dataToTestDeleteFails()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        User user = getNewUser();
+        String username = user.getUsername();
+        return new Object[][]{
+                {username, ResponseCodeConstants.CODE_INTERNAL_ERROR},
+                {username, ResponseCodeConstants.CODE_RESOURCE_NOT_FOUND},
+                {username, ResponseCodeConstants.CODE_NOT_IMPLEMENTED},
+                {username, ResponseCodeConstants.CODE_BAD_REQUEST}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestDeleteFails")
+    public void testDeleteFails(String userName, int expectedScimResponseStatus)
+            throws NotFoundException, NotImplementedException, BadRequestException, CharonException {
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(ENDPOINT_ME);
+
+        if (expectedScimResponseStatus == ResponseCodeConstants.CODE_INTERNAL_ERROR) {
+
```

Comment:
Unnecessary new lines.

Reply:
Fixed in commit a5699e694bc31369fb4a347d4a0a12177e73775d